

<!DOCTYPE html>
<html lang="zh-CN" data-default-color-scheme=&#34;auto&#34;>



<head>
  <meta charset="UTF-8">
  <link rel="apple-touch-icon" sizes="76x76" href="/img/hexo.png">
  <link rel="icon" type="image/png" href="/img/hexo.png">
  <meta name="viewport"
        content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, shrink-to-fit=no">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  
  <meta name="theme-color" content="#2f4154">
  <meta name="description" content="Arita的自留地">
  <meta name="author" content="Arita">
  <meta name="keywords" content="">
  <title>【持续更新】大创的项目进展 &amp; 许许多多莫名其妙吐槽 - Arita-157的观察仪</title>

  <link  rel="stylesheet" href="https://cdn.staticfile.org/twitter-bootstrap/4.4.1/css/bootstrap.min.css" />


  <link  rel="stylesheet" href="https://cdn.staticfile.org/github-markdown-css/4.0.0/github-markdown.min.css" />
  <link  rel="stylesheet" href="/lib/hint/hint.min.css" />

  
    
    <link  rel="stylesheet" href="https://cdn.staticfile.org/highlight.js/10.0.0/styles/github-gist.min.css" />
  

  


<!-- 主题依赖的图标库，不要自行修改 -->

<link rel="stylesheet" href="//at.alicdn.com/t/font_1749284_6peoq002giu.css">



<link rel="stylesheet" href="//at.alicdn.com/t/font_1736178_pjno9b9zyxs.css">


<link  rel="stylesheet" href="/css/main.css" />

<!-- 自定义样式保持在最底部 -->


  <script  src="/js/utils.js" ></script>
  <script  src="/js/color-schema.js" ></script>
<meta name="generator" content="Hexo 5.0.2"></head>


<body>
  <header style="height: 70vh;">
    <nav id="navbar" class="navbar fixed-top  navbar-expand-lg navbar-dark scrolling-navbar">
  <div class="container">
    <a class="navbar-brand"
       href="/">&nbsp;<strong>CosmosClarify</strong>&nbsp;</a>

    <button id="navbar-toggler-btn" class="navbar-toggler" type="button" data-toggle="collapse"
            data-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <div class="animated-icon"><span></span><span></span><span></span></div>
    </button>

    <!-- Collapsible content -->
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav ml-auto text-center">
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/">
                <i class="iconfont icon-home-fill"></i>
                首页
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/archives/">
                <i class="iconfont icon-archive-fill"></i>
                归档
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/categories/">
                <i class="iconfont icon-category-fill"></i>
                分类
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/tags/">
                <i class="iconfont icon-tags-fill"></i>
                标签
              </a>
            </li>
          
        
          
          
          
          
            <li class="nav-item">
              <a class="nav-link" href="/about/">
                <i class="iconfont icon-user-fill"></i>
                关于
              </a>
            </li>
          
        
        
          <li class="nav-item" id="search-btn">
            <a class="nav-link" data-toggle="modal" data-target="#modalSearch">&nbsp;<i
                class="iconfont icon-search"></i>&nbsp;</a>
          </li>
        
        
          <li class="nav-item" id="color-toggle-btn">
            <a class="nav-link" href="javascript:">&nbsp;<i
                class="iconfont icon-dark" id="color-toggle-icon"></i>&nbsp;</a>
          </li>
        
      </ul>
    </div>
  </div>
</nav>

    <div class="banner intro-2" id="background" parallax=true
         style="background: url('https://w.wallhaven.cc/full/96/wallhaven-96719w.jpg') no-repeat center center;
           background-size: cover;">
      <div class="full-bg-img">
        <div class="mask flex-center" style="background-color: rgba(0, 0, 0, 0.3)">
          <div class="container page-header text-center fade-in-up">
            <span class="h2" id="subtitle">
              
            </span>

            
              
  <div class="mt-3 post-meta">
    <i class="iconfont icon-date-fill" aria-hidden="true"></i>
    <time datetime="2020-08-20 12:52" pubdate>
      2020年8月20日 下午
    </time>
  </div>


<div class="mt-1">
  
    
    <span class="post-meta mr-2">
      <i class="iconfont icon-chart"></i>
      3.4k 字
    </span>
  

  
    
    <span class="post-meta mr-2">
      <i class="iconfont icon-clock-fill"></i>
      
      
      16
       分钟
    </span>
  

  
  
</div>

            
          </div>

          
        </div>
      </div>
    </div>
  </header>

  <main>
    
      

<div class="container-fluid">
  <div class="row">
    <div class="d-none d-lg-block col-lg-2"></div>
    <div class="col-lg-8 nopadding-md">
      <div class="container nopadding-md" id="board-ctn">
        <div class="py-5" id="board">
          <article class="post-content mx-auto" id="post">
            <!-- SEO header -->
            <h1 style="display: none">【持续更新】大创的项目进展 &amp; 许许多多莫名其妙吐槽</h1>
            
              <p class="note note-info">
                
                  本文最后更新于：9 天前
                
              </p>
            
            <div class="markdown-body" id="post-body">
              <hr>
<p>我今天就要把我的大创项目拿出来婊.jpg</p>
<a id="more"></a>

<h1 id="2020-09-18"><a href="#2020-09-18" class="headerlink" title="2020.09.18"></a>2020.09.18</h1><p>划水好几天后重新来看看分组：</p>
<ol>
<li><strong>GSE56807+GSE63089</strong>（VALUE类型 RMA signal） 地区都是China，以<em>癌症组织/正常组织</em>作为分组。</li>
<li><strong>GSE100935+GSE34942+GSE35809</strong>（VALUE类型 RMA signal），以<em>样本来源地</em>作为分组</li>
<li><strong>GSE15456+GSE15459+GSE47007</strong>（VALUE类型 MAS5.0 signal），以<em>样本来源地</em>作为分组，**<em>注意GSE15459内部分样本被排除w**</em></li>
<li><strong>GSE118916</strong>（value类型：RMA signal intensity，但是有负数不好判断这到底是个什么东西），<strong>GSE22804</strong>（value类型：lowess- normalized log2 Cy5/Cy3 ratio，貌似可以直接根据ratio值提取差异表达基因），<strong>GSE27342</strong>(value类型：Quantile normalized gene level expression values from PLIER)，这三个set自己和自己分析去吧。</li>
<li><strong>GSE13911和GSE19826</strong>（value类型：MAS5 signal）&amp; <strong>GSE26899和GSE26901</strong>（value类型：Log2 transformed normalized signal intensity），癌症组织和正常组织并不来源于同一位病人，这样的不知道能不能做分析？</li>
<li><strong>GSE110875</strong>（processed data value未知），<strong>GSE30727</strong>（外显子测序），不知道怎么用</li>
</ol>
<p><strong>其他的一些问题：</strong></p>
<ol>
<li>GPL5175 &amp; GPL5188没有gene symbol该怎么用（所以你真的是外显子测序？淦）</li>
<li>学会了Group_lists该怎么往一起合，但是两个样本集的exprSet该怎么往一起合umm data.frame函数又该控制什么条件呢</li>
<li>GSE100935这个样本集里的样本究竟都来自哪个国家……淦，为什么不写的清楚一点！</li>
<li>不知道这个分组到底行不行，sigh</li>
</ol>
<p><strong>本例中涉及到的各种GPL型号：</strong></p>
<ol>
<li><p>GPL570：可以手动注释也可以下载R包注释</p>
</li>
<li><p>GPL11154 ？ 连基本信息都没有 我上当啦！</p>
<blockquote>
<p>测序数据，可以使用refseq/gencode等数据库作为注释文件</p>
</blockquote>
</li>
<li><p>GPL15207：只能手动注释</p>
</li>
<li><p>GPL96：有ID有Gene Symbol，可以手动注释也可以下载R包注释</p>
</li>
<li><p>GPL6480：有ID有GENE_SYMBOL(注：你干嘛要这么写啊啊啊啊啊啊啊)，只能手动注释，注意一下它的名字是GENE_SYMBOL</p>
</li>
<li><p>GPL6947：ID和SYMBOL，手动注释orR包注释（illuminaHumanv3）</p>
</li>
<li><p>GPL5175 ？ 没有GENE SYMBOL也没有R包，存一个网站看看能不能用得到：<a target="_blank" rel="noopener" href="https://www.jianshu.com/p/db1cc9314914">https://www.jianshu.com/p/db1cc9314914</a></p>
<p>愿此生不再遇到GPL5175（信女真诚许愿）</p>
</li>
<li><p>GPL5188：R包注释：huex10sttranscriptcluster</p>
</li>
<li><p>GPL8300：R包注释： hgu95av2</p>
</li>
</ol>
<h1 id="2020-09-10"><a href="#2020-09-10" class="headerlink" title="2020.09.10"></a>2020.09.10</h1><ol>
<li><p>写了一个样本下载.R，根据SampleID.csv下载所需数据（Fine,我大概终于会读.csv表格了？）</p>
</li>
<li><p>根据假期处理GSE19826的经验，写了几个简单的函数（<strong>未证实效性</strong>）用于批处理exprSet表达矩阵，design分组矩阵，contrast.matrix比较矩阵，但是<strong>本项目的重点：将地域性纳入分组条件之一，这个我还没想明白该怎么添进分组条件里去。</strong>也许要从contrast.matrix下手？越复杂的分组条件，比较矩阵相应的也越复杂？（我lay了。</p>
<p>今日碎碎念：有些时候我并不知道一个组长应该怎么做，因为绝大多数情况下我都想方设法自己做完了OTL</p>
</li>
</ol>
<h1 id="2020-09-09"><a href="#2020-09-09" class="headerlink" title="2020.09.09"></a>2020.09.09</h1><ol>
<li><p>明确几个问题：</p>
<ol>
<li><p>处理后的具有两个channel的DNA样本，首列是基因名，表头是样本名，而数据是normalization log2 ratio cy3/cy5（tumor/normal)，癌症组织/正常组织，通过荧光染料标记确定两个样本中存在明显差异表达的基因。</p>
</li>
<li><p>但是双channel的一般是早期测表达的的array数据，不是很建议用XD（此处又过滤掉了两个双channel样本）</p>
</li>
<li><p>用样本来源地分组确认可行。除了limma以外还可以用回归分析筛选差异表达的基因（此处推荐了一个工具<a target="_blank" rel="noopener" href="https://hoffmanlab.org/proj/segtools/">SEGTools</a>。</p>
<p><em>居然是个python软件，fine，这位python朋友我们也不能抛弃:(</em></p>
</li>
</ol>
</li>
<li><p>确认后双channel的GSE22804样本可以用，该样本的数值为normalization log2 ratio cy3/cy5，log2 Ratio&gt;0.58或者log2 Ratio&lt;-1则认为存在差异。<strong>所以我们只需要把数值处在差异区间的基因名称筛出来最后做比较就可以了？</strong></p>
</li>
<li><p>最后把差异基因筛选出来之后应该需要进行GO基因本体分析&amp;KEGG通路分析。</p>
</li>
<li><p>代码进度：把GSE118916进行ID转换后的表达矩阵筛了出来OTL 我在想我是不是应该都筛出来之后往一起合umm</p>
<p><strong>过滤有效探针+筛选最大表达量的探针这一步一定要做，因为检查出来的探针不仅有无效的（不和Gene Symbol联系）甚至还有重的……</strong></p>
</li>
</ol>
<h1 id="2020-09-05"><a href="#2020-09-05" class="headerlink" title="2020.09.05"></a>2020.09.05</h1><ol>
<li>鸽了几天终于把GEO数据挖掘的文档填完了，有几个关于数据的问题&amp;拟了两种分组策略交给师姐看还没有收到回复。虽然还没有正式确定——大概前期准备工作就到这里为止了。</li>
<li>想了想这段时间也没有额外做什么东西。无疑就是读了一点没什么用的中文paper，稍微拟了一下整个项目大概需要做些什么，学了一点点有关GEO数据挖掘的代码，没了。现罹患大创ptsd，只是看到“大创”这两个字我就条件反射性感到焦虑:) </li>
</ol>
<h1 id="2020-09-01"><a href="#2020-09-01" class="headerlink" title="2020.09.01"></a>2020.09.01</h1><ol>
<li>开始补充文档细节，但是还没有挖完。</li>
</ol>
<h1 id="2020-08-31"><a href="#2020-08-31" class="headerlink" title="2020.08.31"></a>2020.08.31</h1><ol>
<li>又双叒叕去GEO上挖了一点数据，但是不知道为什么没有下完（ 垃圾NCBI，每次上NCBI都要挂梯子。</li>
</ol>
<h1 id="2020-08-28"><a href="#2020-08-28" class="headerlink" title="2020.08.28"></a>2020.08.28</h1><ol>
<li>写了一下中期检查（30%）前需要做的事。</li>
</ol>
<h1 id="2020-08-27"><a href="#2020-08-27" class="headerlink" title="2020.08.27"></a>2020.08.27</h1><ol>
<li>和导师对线一天筋疲力尽，但是有了可以带我们的师姐。</li>
<li>和师姐聊了一下一些细节问题，决定还是先继续扒GEO的数据。</li>
</ol>
<h1 id="2020-08-22"><a href="#2020-08-22" class="headerlink" title="2020.08.22"></a>2020.08.22</h1><ol>
<li>用limma包进行分析需要：表达矩阵（exprSet），分组矩阵group_list ，差异比较矩阵contrast.matrix</li>
<li>但是我的差异比较矩阵做的有问题，R不支持空格当rownames/colnames 所以我要把空格换成下划线？还是直接换成control/case组？</li>
</ol>
<h1 id="2020-08-21"><a href="#2020-08-21" class="headerlink" title="2020.08.21"></a>2020.08.21</h1><ol>
<li><p>根据GSE19826和GPL570平台对进行id转换并筛选具有最大表达量的探针</p>
</li>
<li><p>做了一下分组，把癌症组织/非癌症组织分别表现并筛了出来。（据说有利于后面的分析？wtf。</p>
</li>
<li><p>做箱图boxplot，难道接下来就是画图了？(表达矩阵及相关可视化)</p>
</li>
</ol>
<h1 id="2020-08-20"><a href="#2020-08-20" class="headerlink" title="2020.08.20"></a>2020.08.20</h1><ol>
<li><p>用GEOquery下载并生成exprSet和MetaSet</p>
<p><strong>HOWEVER：除了GSE19826同时包含胃癌组织和正常组织外其他芯片内只有胃癌组织的RNA数据，这个要我怎么办 我装死了</strong></p>
</li>
<li><p>根据GSE19826，用hgu133plus2.db筛选有用的探针和没用的探针</p>
<p>结果是41922个可注释探针和12753个不可注释探针</p>
<p>（好了今天鼓捣一天就只到这里，我死了w）</p>
</li>
</ol>
<h1 id="2020-08-19"><a href="#2020-08-19" class="headerlink" title="2020.08.19"></a>2020.08.19</h1><h3 id="GEO数据库相关"><a href="#GEO数据库相关" class="headerlink" title="GEO数据库相关"></a>GEO数据库相关</h3><ol>
<li><p>RAW DATA（不推荐）</p>
</li>
<li><p>Series Matrix Files：表达矩阵 可以读进R（read.table())</p>
<p><em>如何用GEOquery从GEO上下载想要的表达矩阵</em></p>
<p>&lt;!–hexoPostRenderEscape:<pre><code class="hljs R"><span class="hljs-comment">#安装GEOquery</span><br><span class="hljs-keyword">if</span>(!require.namespace(<span class="hljs-string">&quot;BiocManager&quot;</span>,quietly = TURE))<br> install.packages(<span class="hljs-string">&quot;BiocManager&quot;</span>)<br>BiocManager::install(<span class="hljs-string">&quot;GEOquery&quot;</span>)</p>
</li>
</ol>
<p><span class="hljs-comment">#拿gds858举个例子</span><br>options(warn = -<span class="hljs-number">1</span>)<br>supressMessages(<span class="hljs-keyword">library</span>(GEOquery))<br>gds858 &lt;- getGEO(<span class="hljs-string">&#x27;GDS858&#x27;</span>,destdir = <span class="hljs-string">&#x27;.&#x27;</span>)<br>names(Meta(gds858))<span class="hljs-comment">#Meta得到描述信息</span><br>table(Table(gds858))<span class="hljs-comment">#Table得到表达矩阵</span></p>
<p><span class="hljs-comment">#可以用GDS2eSet函数转变为expression set对象</span><br>eset &lt;- GDS2eSet(gds858,do.log2 = <span class="hljs-literal">TRUE</span>)</code></pre>:hexoPostRenderEscape–&gt;</p>
<h3 id="R语言-Limma包的安装流程"><a href="#R语言-Limma包的安装流程" class="headerlink" title="R语言 Limma包的安装流程"></a>R语言 Limma包的安装流程</h3><p>Limma包：Microarray</p>
<p>affy包：affymetrix芯片</p>
<p>lumi包：illumina芯片</p>
<p><strong><em>Limma: Linear Models for Microarray and RNA-Seq Data</em></strong></p>
<p><strong>Bioconductor网站：<a target="_blank" rel="noopener" href="http://www.bioconductor.org/">http://www.bioconductor.org/</a></strong></p>
<p><strong><em>注：用R studio安装貌似会出现一些问题，推荐用R Gui（难道是我没配置镜像站点？Why？）</em></strong></p>
<pre><code class="hljs R"><span class="hljs-keyword">if</span> (!requireNamespace(<span class="hljs-string">&quot;BiocManager&quot;</span>, quietly = <span class="hljs-literal">TRUE</span>))
    install.packages(<span class="hljs-string">&quot;BiocManager&quot;</span>)
BiocManager::install()

<span class="hljs-comment">#安装Limma</span>
BiocManager::install(c(<span class="hljs-string">&quot;limma&quot;</span>))
<span class="hljs-comment">#or</span>
BiocManager::install(<span class="hljs-string">&quot;limma&quot;</span>)

<span class="hljs-comment">#打开Limma Users Guide</span>
<span class="hljs-keyword">library</span>(limma)
limmaUsersGuide()<span class="hljs-comment">#limma包的说明书</span></code></pre>

<h4 id="引用Limma"><a href="#引用Limma" class="headerlink" title="引用Limma"></a>引用Limma</h4><p>using Limma for 差异表达分析：</p>
<p>Phipson, B, Lee, S, Majewski, IJ, Alexander, WS, and Smyth, GK (2016). Robust hyperparameter estimation protects against hypervariable genes and improves power to detect differential expression. <em>Annals of Applied Statistics</em> 10(2), 946–963.</p>
<p>using Limma for RNA序列分析：</p>
<p>Law, CW, Chen, Y, Shi, W, and Smyth, GK (2014). Voom: precision weights unlock linear model analysis tools for RNA-seq read counts. <em>Genome Biology</em> 15, R29.</p>
<h4 id="用Limma包对芯片数据做差异分析"><a href="#用Limma包对芯片数据做差异分析" class="headerlink" title="用Limma包对芯片数据做差异分析"></a>用Limma包对芯片数据做差异分析</h4><p>需要自己做好三个数据：</p>
<p>①表达矩阵 ②分组矩阵 ③差异比较矩阵</p>
<h5 id="基因表达矩阵-读取数据即可"><a href="#基因表达矩阵-读取数据即可" class="headerlink" title="基因表达矩阵 #读取数据即可"></a>基因表达矩阵 #读取数据即可</h5><pre><code class="hljs r"><span class="hljs-comment">#配上GEOquery</span>
<span class="hljs-keyword">library</span>(GEOquery)
gse19826 &lt;- getGEO(<span class="hljs-string">&#x27;GSE19826&#x27;</span>,destdir = <span class="hljs-string">&#x27;.&#x27;</span>)
<span class="hljs-keyword">library</span>(limma)
exprSet = read.table(<span class="hljs-string">&quot;GSE19826_series_matrix.txt.gz&quot;</span>,
                     comment.char = <span class="hljs-string">&quot;!&quot;</span>,
                     stringsAsFactors = <span class="hljs-literal">F</span>,
                     header = <span class="hljs-literal">T</span>)
<span class="hljs-comment">#Q：read.table的参数</span>
rownames(exprSet) = exprSet[,<span class="hljs-number">1</span>]
exprSet = exprSet[,-<span class="hljs-number">1</span>]

<span class="hljs-comment">#以上全部作废</span>


downGEO &lt;-<span class="hljs-keyword">function</span>(studyID,dest_dir)
&#123;
    <span class="hljs-keyword">library</span>(GEOquery)
    estr &lt;-getGEO(GEO = studyID,
                 destdir = dest_dir,
                 getGPL = <span class="hljs-literal">TRUE</span>)
    expr = exprs(estr[[<span class="hljs-number">1</span>]])
    pdata = pdata(estr[[<span class="hljs-number">1</span>]])

    write.csv(expr,paste0(studyID,<span class="hljs-string">&quot;_exprSet.csv&quot;</span>))
   write.csv(pdata,paste0(studyID,<span class="hljs-string">&quot;_metadata.csv&quot;</span>))

    <span class="hljs-keyword">return</span>(estr)
&#125;
downGEO(studyID =,dest_dir = <span class="hljs-string">&#x27;.&#x27;</span>)
<span class="hljs-comment">#studyID就是你要下载的数据的GEO号，后面的不用管了，可以自动在文件存放目录生成表达矩阵和信息谱两种</span>

<span class="hljs-comment">#进行ID转换 以GSE19826为例</span>

<span class="hljs-keyword">library</span>(GEOquery)

<span class="hljs-comment">#一个返回exprSet的函数 可能没啥用</span>
get_exprSet &lt;- <span class="hljs-keyword">function</span>(studyID,dest_dir)&#123;
  e_Set &lt;- getGEO(GEO = studyID,
                  destdir = <span class="hljs-string">&#x27;.&#x27;</span>,
                  getGPL = <span class="hljs-literal">F</span>)
  expr_set = exprs(e_Set[[<span class="hljs-number">1</span>]])
  <span class="hljs-keyword">return</span>(expr_set)
&#125;
downGEO(studyID = <span class="hljs-string">&#x27;GSE19826&#x27;</span>,dest_dir = <span class="hljs-string">&#x27;.&#x27;</span>)
exprSet = get_exprSet(studyID = <span class="hljs-string">&#x27;GSE19826&#x27;</span>,
                      dest_dir = <span class="hljs-string">&#x27;.&#x27;</span>)

<span class="hljs-comment">#下面进行id转换</span>
<span class="hljs-comment">#载入包</span>
<span class="hljs-keyword">library</span>(hgu133plus2.db)
ids = toTable(hgu133plus2SYMBOL)

<span class="hljs-comment">#①筛选有用的探针和没用的探针</span>

table(rownames(exprSet) %<span class="hljs-keyword">in</span>% ids$probe_id)
exprSet = exprSet[rownames(exprSet)%<span class="hljs-keyword">in</span>% ids$probe_id]

<span class="hljs-comment">#②将没用的探针筛掉 再选择有最大表达量的探针</span>
ids = ids[match(rownames(exprSet),ids$probe_id),]
tmp = by(exprSet,
        ids$symbol,
        <span class="hljs-keyword">function</span>(x) rownames(x)[which.max(rowMeans(x))])
         )
probes = as.character(tmp)
exprSet = exprSet[rownames(exprSet) %<span class="hljs-keyword">in</span>% probes,]
dim(exprSet)
rownames(exprSet) = ids[match(rownames(exprSet),ids$probe_id),<span class="hljs-number">2</span>]

<span class="hljs-comment">#boxplot的一个画法</span>
exprSet[<span class="hljs-string">&#x27;PAX8&#x27;</span>,]
boxplot(exprSet[<span class="hljs-string">&#x27;PAX8&#x27;</span>,])

<span class="hljs-comment">#做分组，将癌症组织/非癌症组织分出来</span>
<span class="hljs-comment">#首先把group_list分组条件做出来</span>
get_pData &lt;- <span class="hljs-keyword">function</span>(studyID,dest_dir)&#123;
  <span class="hljs-keyword">library</span>(GEOquery)
  a &lt;- getGEO(GEO = studyID,destdir = dest_dir)
  pData = pData(a[[<span class="hljs-number">1</span>]])
  <span class="hljs-keyword">return</span>(pData)
&#125;
pData = get_pData(studyID = <span class="hljs-string">&#x27;GSE19826&#x27;</span>,
                  dest_dir = <span class="hljs-string">&#x27;.&#x27;</span>)
<span class="hljs-comment">#pData的第39列为癌症组织/非癌症组织的信息</span>
group_list = as.character(pData[,<span class="hljs-number">39</span>])

<span class="hljs-comment">#正式进行分组</span>
<span class="hljs-keyword">library</span>(reshape2)
exprSet_L = melt(exprSet)
colnames(exprSet_L) = c(<span class="hljs-string">&#x27;probe&#x27;</span>,<span class="hljs-string">&#x27;sample&#x27;</span>,<span class="hljs-string">&#x27;value&#x27;</span>)
exprSet_L$group = rep(group_list,each = nrow(exprSet))
head(exprSet_L)

<span class="hljs-comment">#载入limma包</span>
<span class="hljs-comment">#制作分组矩阵</span>
<span class="hljs-keyword">library</span>(limma)
design &lt;- model.matrix(~<span class="hljs-number">0</span> + factor(group_list))
colnames(design) = levels(factor(group_list))
rownames(design) = colnames(exprSet)
</code></pre>

<p><img src="C:\Users\Administrator\Documents\Rplot.png" srcset="/img/loading.gif" alt="Rplot"></p>
<p><em>注：PAX8基因在这些芯片数据中的boxplot如图所示</em></p>
<h3 id="ID转换-读取探针并注释"><a href="#ID转换-读取探针并注释" class="headerlink" title="ID转换 *读取探针并注释"></a>ID转换 *读取探针并注释</h3><p><strong>平台与注释包的对应：</strong></p>
<p>1    <strong>GPL32</strong>    <strong>mgu74a</strong>    [MG_U74A] Affymetrix Murine Genome U74A Array<br>2    <strong>GPL33</strong>    <strong>mgu74b</strong>    [MG_U74B] Affymetrix Murine Genome U74B Array<br>3    <strong>GPL34    mgu74c</strong>    [MG_U74C] Affymetrix Murine Genome U74C Array<br>4    <strong>GPL71    ag</strong>    [AG] Affymetrix Arabidopsis Genome Array<br>5    <strong>GPL72    drosgenome1</strong>    [DrosGenome1] Affymetrix Drosophila Genome Array<br>6    <strong>GPL74    hcg110</strong>    [HC_G110] Affymetrix Human Cancer Array<br>7    <strong>GPL75    mu11ksuba</strong>    [Mu11KsubA] Affymetrix Murine 11K SubA Array<br>8    <strong>GPL76    mu11ksubb</strong>    [Mu11KsubB] Affymetrix Murine 11K SubB Array<br>9    <strong>GPL77    mu19ksuba</strong>    [Mu19KsubA] Affymetrix Murine 19K SubA Array<br>10    <strong>GPL78    mu19ksubb</strong>    [Mu19KsubB] Affymetrix Murine 19K SubB Array<br>11    <strong>GPL79    mu19ksubc</strong>    [Mu19KsubC] Affymetrix Murine 19K SubC Array<br>12    <strong>GPL80    hu6800</strong>    [Hu6800] Affymetrix Human Full Length HuGeneFL Array<br>13    <strong>GPL81    mgu74av2</strong>    [MG_U74Av2] Affymetrix Murine Genome U74A Version 2 Array<br>14    <strong>GPL82    mgu74bv2</strong>    [MG_U74Bv2] Affymetrix Murine Genome U74B Version 2 Array<br>15    <strong>GPL83    mgu74cv2</strong>    [MG_U74Cv2] Affymetrix Murine Genome U74 Version 2 Array<br>16    <strong>GPL85    rgu34a</strong>    [RG_U34A] Affymetrix Rat Genome U34 Array<br>17    <strong>GPL86    rgu34b</strong>    [RG_U34B] Affymetrix Rat Genome U34 Array<br>18    <strong>GPL87    rgu34c</strong>    [RG_U34C] Affymetrix Rat Genome U34 Array<br>19    <strong>GPL88    rnu34</strong>    [RN_U34] Affymetrix Rat Neurobiology U34 Array<br>20    <strong>GPL89    rtu34</strong>    [RT_U34] Affymetrix Rat Toxicology U34 Array<br>21    <strong>GPL90    ygs98</strong>    [YG_S98] Affymetrix Yeast Genome S98 Array<br>22    <strong>GPL91    hgu95av2</strong>    [HG_U95A] Affymetrix Human Genome U95A Array<br>23    <strong>GPL92    hgu95b</strong>    [HG_U95B] Affymetrix Human Genome U95B Array<br>24    <strong>GPL93    hgu95c</strong>    [HG_U95C] Affymetrix Human Genome U95C Array<br>25    <strong>GPL94    hgu95d</strong>    [HG_U95D] Affymetrix Human Genome U95D Array<br>26    <strong>GPL95    hgu95e</strong>    [HG_U95E] Affymetrix Human Genome U95E Array<br>27    <strong>GPL96    hgu133a</strong>    [HG-U133A] Affymetrix Human Genome U133A Array<br>28    <strong>GPL97    hgu133b</strong>    [HG-U133B] Affymetrix Human Genome U133B Array<br>29    <strong>GPL98    hu35ksuba</strong>    [Hu35KsubA] Affymetrix Human 35K SubA Array<br>30    <strong>GPL99    hu35ksubb</strong>    [Hu35KsubB] Affymetrix Human 35K SubB Array<br>31    <strong>GPL100    hu35ksubc</strong>    [Hu35KsubC] Affymetrix Human 35K SubC Array<br>32    <strong>GPL101    hu35ksubd</strong>    [Hu35KsubD] Affymetrix Human 35K SubD Array<br>33    <strong>GPL198    ath1121501</strong>    [ATH1-121501] Affymetrix Arabidopsis ATH1 Genome Array<br>34    <strong>GPL199    ecoli2</strong>    [Ecoli_ASv2] Affymetrix E. coli Antisense Genome Array<br>35    <strong>GPL200    celegans</strong>    [Celegans] Affymetrix C. elegans Genome Array<br>36    <strong>GPL201    hgfocus</strong>    [HG-Focus] Affymetrix Human HG-Focus Target Array<br>37    <strong>GPL339    moe430a</strong>    [MOE430A] Affymetrix Mouse Expression 430A Array<br>38    <strong>GPL340    mouse4302</strong>    [MOE430B] Affymetrix Mouse Expression 430B Array<br>39    <strong>GPL341    rae230a</strong>    [RAE230A] Affymetrix Rat Expression 230A Array<br>40    <strong>GPL342    rae230b</strong>    [RAE230B] Affymetrix Rat Expression 230B Array<br>41    <strong>GPL570    hgu133plus2</strong>    [HG-U133_Plus_2] Affymetrix Human Genome U133 Plus 2.0 Array<br>42    <strong>GPL571    hgu133a2</strong>    [HG-U133A_2] Affymetrix Human Genome U133A 2.0 Array<br>43    <strong>GPL886    hgug4111a</strong>    Agilent-011871 Human 1B Microarray G4111A (Feature Number version)<br>44    <strong>GPL887    hgug4110b</strong>    Agilent-012097 Human 1A Microarray (V2) G4110B (Feature Number version)<br>45    <strong>GPL1261    mouse430a2</strong>    [Mouse430_2] Affymetrix Mouse Genome 430 2.0 Array<br>46    <strong>GPL1318    xenopuslaevis</strong>    [Xenopus_laevis] Affymetrix Xenopus laevis Genome Array<br>47    <strong>GPL1319    zebrafish</strong>    [Zebrafish] Affymetrix Zebrafish Genome Array<br>48    <strong>GPL1322    drosophila2</strong>    [Drosophila_2] Affymetrix Drosophila Genome 2.0 Array<br>49    <strong>GPL1352    u133x3p</strong>    [U133_X3P] Affymetrix Human X3P Array<br>50    <strong>GPL1355    rat2302</strong>    [Rat230_2] Affymetrix Rat Genome 230 2.0 Array<br>51    <strong>GPL1708    hgug4112a</strong>    Agilent-012391 Whole Human Genome Oligo Microarray G4112A (Feature Number version)<br>52    <strong>GPL2112    bovine</strong>    [Bovine] Affymetrix Bovine Genome Array<br>53    <strong>GPL2529    yeast2</strong>    [Yeast_2] Affymetrix Yeast Genome 2.0 Array<br>54    <strong>GPL2891    h20kcod</strong>    GE Healthcare/Amersham Biosciences CodeLinkダｽ UniSet Human 20K I Bioarray<br>55    <strong>GPL2898    adme16cod</strong>    GE Healthcare/Amersham Biosciences CodeLinkダｽ ADME Rat 16-Assay Bioarray<br>56    <strong>GPL3154    ecoli2</strong>    [E_coli_2] Affymetrix E. coli Genome 2.0 Array<br>57    <strong>GPL3213    chicken</strong>    [Chicken] Affymetrix Chicken Genome Array<br>58    <strong>GPL3533    porcine</strong>    [Porcine] Affymetrix Porcine Genome Array<br>59    <strong>GPL3738    canine2</strong>    [Canine_2] Affymetrix Canine Genome 2.0 Array<br>60    <strong>GPL3921    hthgu133a</strong>    [HT_HG-U133A] Affymetrix HT Human Genome U133A Array<br>61    <strong>GPL3979    canine</strong>    [Canine] Affymetrix Canine Genome 1.0 Array<br>62    <strong>GPL4032</strong>        [Maize] Affymetrix Maize Genome Array<br>63    <strong>GPL4191    h10kcod</strong>    CodeLink UniSet Human I Bioarray<br>64    <strong>GPL5188    huex10sttranscriptcluster</strong>    [HuEx-1_0-st] Affymetrix Human Exon 1.0 ST Array [probe set (exon) version]<br>65    <strong>GPL5689    hgug4100a</strong>    Agilent Human 1 cDNA Microarray (G4100A) [layout C]<br>66    <strong>GPL6097    illuminaHumanv1</strong>    Illumina human-6 v1.0 expression beadchip<br>67    <strong>GPL6102    illuminaHumanv2</strong>    Illumina human-6 v2.0 expression beadchip<br>68    <strong>GPL6244    hugene10sttranscriptcluster</strong>    [HuGene-1_0-st] Affymetrix Human Gene 1.0 ST Array [transcript (gene) version]<br>69    <strong>GPL6246    mogene10sttranscriptcluster</strong>    [MoGene-1_0-st] Affymetrix Mouse Gene 1.0 ST Array [transcript (gene) version]<br>70    <strong>GPL6885    illuminaMousev2</strong>    Illumina MouseRef-8 v2.0 expression beadchip<br>71    <strong>GPL6947    illuminaHumanv3</strong>    Illumina HumanHT-12 V3.0 expression beadchip<br>72    <strong>GPL8300    hgu95av2</strong>    [HG_U95Av2] Affymetrix Human Genome U95 Version 2 Array<br>73    <strong>GPL8321    mouse430a2</strong>    [Mouse430A_2] Affymetrix Mouse Genome 430A 2.0 Array<br>74    <strong>GPL8490    IlluminaHumanMethylation27k</strong>    Illumina HumanMethylation27 BeadChip (HumanMethylation27_270596_v.1.2)<br>75    <strong>GPL10558    illuminaHumanv4</strong>    Illumina HumanHT-12 V4.0 expression beadchip<br>76    <strong>GPL11532    hugene11sttranscriptcluster</strong>    [HuGene-1_1-st] Affymetrix Human Gene 1.1 ST Array [transcript (gene) version]<br>77    <strong>GPL13497    HsAgilentDesign026652</strong>    Agilent-026652 Whole Human Genome Microarray 4x44K v2 (Probe Name version)<br>78    <strong>GPL13534    IlluminaHumanMethylation450k</strong>    Illumina HumanMethylation450 BeadChip (HumanMethylation450_15017482)<br>79    <strong>GPL13667    hgu219</strong>    [HG-U219] Affymetrix Human Genome U219 Array<br>80    <strong>GPL14877    hgu133plus2</strong>    Affymetrix Human Genome U133 Plus 2.0 Array [Brainarray Version 13, HGU133Plus2_Hs_ENTREZG]<br>81    <strong>GPL15380    GGHumanMethCancerPanelv1</strong>    Illumina Sentrix Array Matrix (SAM) - GoldenGate Methylation Cancer Panel I<br>82    <strong>GPL15396    hthgu133b</strong>    [HT_HG-U133B] Affymetrix HT Human Genome U133B Array [custom CDF: ENTREZ brainarray v. 14]<br>83    <strong>GPL17556    hugene10sttranscriptcluster</strong>    [HuGene-1_0-st] Affymetrix Human Gene 1.0 ST Array [HuGene10stv1_Hs_ENTREZG_17.0.0]<br>84    <strong>GPL17897    hthgu133a</strong>    [HT_HG-U133A] Affymetrix Human Genome U133A Array (custom CDF: HTHGU133A_Hs_ENTREZG.cdf version 17.0.0)<br>85    <strong>GPL18190    hugene11sttranscriptcluster</strong>    [HuGene-1_1-st] Affymetrix Human Gene 1.1 ST Array [CDF: Brainarray HuGene11stv1_Hs_ENTREZG_15.1.0]</p>

            </div>
            <hr>
            <div>
              <div class="post-metas mb-3">
                
                  <div class="post-meta mr-3">
                    <i class="iconfont icon-category"></i>
                    
                      <a class="hover-with-bg" href="/categories/%E5%A4%A7%E5%88%9B/">大创</a>
                    
                  </div>
                
                
                  <div class="post-meta">
                    <i class="iconfont icon-tags"></i>
                    
                      <a class="hover-with-bg" href="/tags/%E9%A1%B9%E7%9B%AE%E8%BF%9B%E5%B1%95%E8%AE%B0%E5%BD%95/">项目进展记录</a>
                    
                  </div>
                
              </div>
              
              
                <div class="post-prevnext row">
                  <article class="post-prev col-6">
                    
                    
                      <a href="/2020/09/02/20200902-%E3%80%90DAYLOG%E3%80%91%E5%9B%9EHUST%E4%BB%A5%E6%9D%A5%E6%AF%8F%E4%B8%80%E5%A4%A9%E7%9A%84%E8%AE%B0%E5%BD%95/">
                        <i class="iconfont icon-arrowleft"></i>
                        <span class="hidden-mobile">【DAYLOG】回HUST以来每一天的记录</span>
                        <span class="visible-mobile">上一篇</span>
                      </a>
                    
                  </article>
                  <article class="post-next col-6">
                    
                    
                      <a href="/2020/08/20/2020-08-20-%E7%BD%AE%E9%A1%B6/">
                        <span class="hidden-mobile">置顶 2020/09/18</span>
                        <span class="visible-mobile">下一篇</span>
                        <i class="iconfont icon-arrowright"></i>
                      </a>
                    
                  </article>
                </div>
              
            </div>

            
          </article>
        </div>
      </div>
    </div>
    
      <div class="d-none d-lg-block col-lg-2 toc-container" id="toc-ctn">
        <div id="toc">
  <p class="toc-header"><i class="iconfont icon-list"></i>&nbsp;目录</p>
  <div id="tocbot"></div>
</div>

      </div>
    
  </div>
</div>

<!-- Custom -->


    
  </main>

  
    <a id="scroll-top-button" href="#" role="button">
      <i class="iconfont icon-arrowup" aria-hidden="true"></i>
    </a>
  

  
    <div class="modal fade" id="modalSearch" tabindex="-1" role="dialog" aria-labelledby="ModalLabel"
     aria-hidden="true">
  <div class="modal-dialog modal-dialog-scrollable modal-lg" role="document">
    <div class="modal-content">
      <div class="modal-header text-center">
        <h4 class="modal-title w-100 font-weight-bold">搜索</h4>
        <button type="button" id="local-search-close" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body mx-3">
        <div class="md-form mb-5">
          <input type="text" id="local-search-input" class="form-control validate">
          <label data-error="x" data-success="v"
                 for="local-search-input">关键词</label>
        </div>
        <div class="list-group" id="local-search-result"></div>
      </div>
    </div>
  </div>
</div>
  

  
    <!-- APlayer 音乐播放器 -->
    <div id="aplayer"></div>
    <script defer src="https://cdn.staticfile.org/aplayer/1.10.1/APlayer.min.js" ></script>
<link  rel="stylesheet" href="https://cdn.staticfile.org/aplayer/1.10.1/APlayer.min.css" />
<script type="text/javascript">
  var oldLoadAp = window.onload;
  window.onload = function () {
    oldLoadAp && oldLoadAp();

    new APlayer({
      container: document.getElementById('aplayer'),
      fixed: true,
      autoplay: 'false' === 'true',
      loop: 'all',
      order: 'random',
      theme: '#b7daff',
      preload: 'none',
      audio: [{"name":"Jedi Temple March","artist":"Samuel Kim Music","url":"/songs/March.mp3","cover":"/img/hqdefault.webp"},{"name":"Born To Die","artist":"Lana Del Rey","url":"/songs/Lana Del Rey - Born To Die (Album Version).mp3","cover":"/img/img2.jpg"},{"name":"Shallow","artist":"Carys Selvey","url":"/songs/Carys Selvey - Shallow.mp3","cover":"/img/20161221050218301276.jpg"},{"name":"Ignite","artist":"Alan Walker","url":"/songs/Ignite.mp3","cover":"/img/Ignite.jpg"}]
    });
  }
</script>

  

  

  <footer class="mt-5">
  <div class="text-center py-3">
    <div>
      <a href="https://hexo.io" target="_blank" rel="nofollow noopener"><span>Hexo</span></a>
      <i class="iconfont icon-love"></i>
      <a href="https://github.com/fluid-dev/hexo-theme-fluid" target="_blank" rel="nofollow noopener">
        <span>Fluid</span></a>
    </div>
    

    

    
  </div>
</footer>

<!-- SCRIPTS -->
<script  src="https://cdn.staticfile.org/jquery/3.4.1/jquery.min.js" ></script>
<script  src="https://cdn.staticfile.org/twitter-bootstrap/4.4.1/js/bootstrap.min.js" ></script>
<script  src="/js/debouncer.js" ></script>
<script  src="/js/main.js" ></script>

<!-- Plugins -->


  
    <script  src="/js/lazyload.js" ></script>
  



  <script defer src="https://cdn.staticfile.org/clipboard.js/2.0.6/clipboard.min.js" ></script>
  <script  src="/js/clipboard-use.js" ></script>







  <script  src="https://cdn.staticfile.org/tocbot/4.11.1/tocbot.min.js" ></script>
  <script>
    $(document).ready(function () {
      var boardCtn = $('#board-ctn');
      var boardTop = boardCtn.offset().top;

      tocbot.init({
        tocSelector: '#tocbot',
        contentSelector: '#post-body',
        headingSelector: 'h1,h2,h3,h4,h5,h6',
        linkClass: 'tocbot-link',
        activeLinkClass: 'tocbot-active-link',
        listClass: 'tocbot-list',
        isCollapsedClass: 'tocbot-is-collapsed',
        collapsibleClass: 'tocbot-is-collapsible',
        collapseDepth: 0,
        scrollSmooth: true,
        headingsOffset: -boardTop
      });
      if ($('.toc-list-item').length > 0) {
        $('#toc').css('visibility', 'visible');
      }
    });
  </script>



  <script  src="https://cdn.staticfile.org/typed.js/2.0.11/typed.min.js" ></script>
  <script>
    var typed = new Typed('#subtitle', {
      strings: [
        '  ',
        "【持续更新】大创的项目进展 & 许许多多莫名其妙吐槽&nbsp;",
      ],
      cursorChar: "_",
      typeSpeed: 70,
      loop: false,
    });
    typed.stop();
    $(document).ready(function () {
      $(".typed-cursor").addClass("h2");
      typed.start();
    });
  </script>



  <script  src="https://cdn.staticfile.org/anchor-js/4.2.2/anchor.min.js" ></script>
  <script>
    anchors.options = {
      placement: "right",
      visible: "hover",
      
    };
    var el = "h1,h2,h3,h4,h5,h6".split(",");
    var res = [];
    for (item of el) {
      res.push(".markdown-body > " + item)
    }
    anchors.add(res.join(", "))
  </script>



  <script  src="/js/local-search.js" ></script>
  <script>
    var path = "/local-search.xml";
    var inputArea = document.querySelector("#local-search-input");
    inputArea.onclick = function () {
      searchFunc(path, 'local-search-input', 'local-search-result');
      this.onclick = null
    }
  </script>



  <script  src="https://cdn.staticfile.org/fancybox/3.5.7/jquery.fancybox.min.js" ></script>
  <link  rel="stylesheet" href="https://cdn.staticfile.org/fancybox/3.5.7/jquery.fancybox.min.css" />

  <script>
    $('#post img:not(.no-zoom img, img[no-zoom]), img[zoom]').each(
      function () {
        var element = document.createElement('a');
        $(element).attr('data-fancybox', 'images');
        $(element).attr('href', $(this).attr('src'));
        $(this).wrap(element);
      }
    );
  </script>




















</body>
</html>
